syntax = "proto3";

package dataserver;

option cc_generic_services = true; 

message AddResponse {
    bool success = 1;
}

message AddRequest {
    string file_name = 1;
    bytes contents = 2;
}

message GetResponse {
    bool success = 1;
    bytes contents = 2;
}

message GetRequest {
    string file_name = 1;
}

message DelResponse {
    bool success = 1;
}

message DelRequest {
    string file_name = 1;
}

service DataService {
    rpc Add(AddRequest) returns (AddResponse);
    rpc Get(GetRequest) returns (GetResponse);
    rpc Del(DelRequest) returns (DelResponse);
}
